File: rb/features.rb
# IsEven SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IsEvenFeatures
  def self.make_feature(name)
    case name
    when "base"
      IsEvenBaseFeature.new
    when "test"
      IsEvenTestFeature.new
    else
      IsEvenBaseFeature.new
    end
  end
end

File: py/features.py
# IsEven SDK feature factory

from feature.base_feature import IsEvenBaseFeature
from feature.test_feature import IsEvenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IsEvenBaseFeature(),
        "test": lambda: IsEvenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
